cdap-proto/src/main/java/co/cask/cdap/proto/id/ApplicationId.java

```diff
@@ -30,12 +30,18 @@
 public class ApplicationId extends EntityId implements NamespacedId, ParentedId<NamespaceId> {
   private final String namespace;
   private final String application;
+  private final String version;
   private transient Integer hashCode;
 
-  public ApplicationId(String namespace, String application) {
+  public ApplicationId(String namespace, String application, String version) {
     super(EntityType.APPLICATION);
     this.namespace = namespace;
     this.application = application;
+    this.version = version;
+  }
+
+  public ApplicationId(String namespace, String application) {
+    this(namespace, application, "-SNAPSHOT");
```

Comment:
nope, we are just going to check for -SNAPSHOT suffix to see whether that version is immutable or not. If no version is provided, the default one should be mutable to adhere to the current capability.